.github/workflows/nix.yml

```diff
@@ -26,3 +23,41 @@ jobs:
 
       - name: Build all flake outputs
         run: om ci
+
+      - name: Build and push Docker image
+        if: github.event_name == 'push' && github.event.pull_request.merged == true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -euo pipefail
+
+          ARG1="${1:-dockerImage}"
+          shift 1 || true
+
+          set -x
+          # Show git status as Docker tag is based on the working copy status
+          git status
+
+          # Build the flake output and load the Docker image
+          docker load -i "$(nix build .#"$ARG1" --print-out-paths --no-update-lock-file)"
+
+          # Fetch the image name and tag
+          IMAGE_NAME="$(nix eval --json .#packages.x86_64-linux."$ARG1".buildArgs | jq -r '"\(.name):\(.tag)"')"
+          echo "Built and loaded: ${IMAGE_NAME}"
+
+          # Use a temporary directory for HOME to avoid insecure password storage
+          HOME="$(mktemp -d)"
+          export HOME
+
+          # Ensure the Docker image is removed after pushing
+          trap 'docker rmi "${IMAGE_NAME}"; rm -rf "$HOME"' EXIT
+
+          # Push the Docker image
+          echo "$GITHUB_TOKEN" | docker login ghcr.io -u "${{ github.actor }}" --password-stdin
+          docker tag "ghcr.io/nammayatri/beckn-gateway:a7c614" "$IMAGE_NAME"
+          docker push "$IMAGE_NAME"
+
+          echo "Pushed image: ${IMAGE_NAME}"
+
+          # Log out from Docker registry
+          docker logout ghcr.io
```

Comment:
Ah, I just realized something. I think we should do the entire docker login and push from a `ubuntu-latest` runner, so we don't have to deal with these security concerns. You can use `actions/upload-artifact` and `actions/download-artifact` - have Nix upload the image tarball, and download it from a ubuntu-latest runner, loading and pushing it.

See 

https://github.com/juspay/omnix/blob/1f62a25f92c84ce88029783ad5a1bad1245edec6/.github/workflows/ci.yaml#L66

https://github.com/juspay/omnix/blob/1f62a25f92c84ce88029783ad5a1bad1245edec6/.github/workflows/ci.yaml#L95

Reply:
hey @srid made changes regarding this